src/java.base/share/classes/jdk/classfile/attribute/ModuleExportInfo.java

```diff
@@ -121,4 +125,56 @@ static ModuleExportInfo of(PackageEntry exports,
                                ModuleEntry... exportsTo) {
         return of(exports, Util.flagsToBits(AccessFlag.Location.MODULE_EXPORTS, exportFlags), exportsTo);
     }
+
+    /**
+     * {@return a module export description}
+     * @param exports the exported package
+     * @param exportFlags the export flags
+     * @param exportsTo the modules to which this package is exported
+     */
+    static ModuleExportInfo of(PackageDesc exports,
+                               int exportFlags,
+                               List<ModuleDesc> exportsTo) {
+        return new UnboundAttribute.UnboundModuleExportInfo(TemporaryConstantPool.INSTANCE.packageEntry(TemporaryConstantPool.INSTANCE.utf8Entry(exports.packageName())),
+                exportFlags,
+                exportsTo.stream().map(e -> TemporaryConstantPool.INSTANCE.moduleEntry(TemporaryConstantPool.INSTANCE.utf8Entry(e.moduleName()))).collect(Collectors.toList()));
```

Comment:
Direct factory methods ::moduleEntry(ModuleDesc) and ::packageEntry(PackageDesc) are available and direct collector method Stream::toList():
```
return new UnboundAttribute.UnboundModuleExportInfo(TemporaryConstantPool.INSTANCE.packageEntry(exports),
                 exportFlags,
                 exportsTo.stream().map(TemporaryConstantPool.INSTANCE::moduleEntry).toList());
```